domains/pallets/operator-rewards/src/lib.rs

```diff
@@ -61,14 +80,93 @@ mod pallet {
+            BlockRewards::<T>::set(Default::default());
             T::DbWeight::get().writes(1)
         }
+
+        fn on_finalize(_now: BlockNumberFor<T>) {
+            let transaction_byte_fee = NextDomainTransactionByteFee::<T>::get();
+            DomainTransactionByteFee::<T>::put(transaction_byte_fee);
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight((
+            // TODO: proper weight
+			Weight::from_all(10_000),
+			DispatchClass::Mandatory
+		))]
+        pub fn set_next_domain_transaction_byte_fee(
+            origin: OriginFor<T>,
+            #[pallet::compact] transaction_byte_fee: T::Balance,
+        ) -> DispatchResult {
+            ensure_none(origin)?;
+            NextDomainTransactionByteFee::<T>::put(transaction_byte_fee);
+            Ok(())
+        }
+    }
+
+    #[pallet::inherent]
+    impl<T: Config> ProvideInherent for Pallet<T> {
+        type Call = Call<T>;
+        type Error = InherentError;
+        const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
+
+        fn create_inherent(data: &InherentData) -> Option<Self::Call> {
+            let inherent_data = data
+                .get_data::<InherentType>(&INHERENT_IDENTIFIER)
+                .expect("Operator rewards inherent data not correctly encoded")
+                .expect("Operator rewards inherent data must be provided");
+
+            let transaction_byte_fee = inherent_data.saturated_into::<T::Balance>();
+
+            Some(Call::set_next_domain_transaction_byte_fee {
+                transaction_byte_fee,
+            })
+        }
+
+        fn check_inherent(
+            call: &Self::Call,
+            data: &InherentData,
+        ) -> result::Result<(), Self::Error> {
+            let inherent_data = data
+                .get_data::<InherentType>(&INHERENT_IDENTIFIER)
+                .expect("Operator rewards inherent data not correctly encoded")
+                .expect("Operator rewards inherent data must be provided");
+
+            let provided_transaction_byte_fee = inherent_data.saturated_into::<T::Balance>();
+
+            if let Call::set_next_domain_transaction_byte_fee {
+                transaction_byte_fee,
+            } = call
+            {
+                if transaction_byte_fee != &provided_transaction_byte_fee {
+                    return Err(InherentError::IncorrectDomainTransactionByteFee);
+                }
+            }
+
+            Ok(())
+        }
+
+        fn is_inherent(call: &Self::Call) -> bool {
+            matches!(call, Call::set_next_domain_transaction_byte_fee { .. })
+        }
     }
 
     impl<T: Config> Pallet<T> {
         pub fn note_operator_rewards(rewards: T::Balance) {
-            let next_block_rewards = BlockRewards::<T>::get().saturating_add(rewards);
+            let mut next_block_rewards = BlockRewards::<T>::get();
+            next_block_rewards.execution_fee =
```

Comment:
I'm confused with this name here. Why is this `next_block_reward` ? These are the fees from the current domain block state transition correct ?

Reply:
The `next` refers to the next value after this call, yeah it is quite confusing when using along with `block`, how about changing it to `new_xx`?